src/libsyntax/feature_gate.rs

```diff
@@ -2446,6 +2449,26 @@ pub fn check_crate(krate: &ast::Crate,
         parse_sess: sess,
         plugin_attributes,
     };
+
+    krate
+        .module
+        .items
+        .iter()
+        .filter_map(|i| if let ast::ItemKind::Fn(decl, ..) = &i.node {
+            Some(&decl.inputs)
+        }
+        else {
+            None
+        })
+        .flatten()
+        .flat_map(|i| i.attrs.as_ref())
+        .for_each(|a| gate_feature!(
+            &ctx,
+            param_attrs,
+            a.span,
+            "attributes on formal function parameters are unstable"
+        ));
```

Comment:
I suspect that this code is trying to mimic https://github.com/rust-lang/rust/commit/7d5c29b9eae5857c040bf6f1b2d729596c8af3ae, but is missing the point.

A field `param_attr_spans: RefCell<Vec<Span>>` needs to be added to `struct ParseSess` and spans need to be added to it during parsing.
Then code here in `feature_gate.rs` needs to loop through `param_attr_spans` and report feature errors.